lib/praxis/trait.rb

```diff
@@ -68,17 +68,20 @@ def describe
         desc[:routing] = ConfigHash.new(&@routing).to_hash
       end
 
-      @attribute_groups.each_with_object(desc) do |(name, block), hash|
+      @attribute_groups.each_with_object(desc) do |(name, blocks), hash|
         type_class = if name == :headers
           # Headers are special:
           # Keys are strings, they have a special DSL, and are case insensitive
           hash_opts = {
             dsl_compiler: ActionDefinition::HeadersDSLCompiler,
             case_insensitive_load: true
           }
-          Attributor::Hash.of(key: String).construct(block, hash_opts)
+          Attributor::Hash.of(key: String).construct(Proc.new {}, hash_opts)
         else
-          Attributor::Hash.construct(block)
+          Attributor::Hash.construct(Proc.new {})
+        end
+        blocks.each do |block|
+          type_class.construct(block)
```

Comment:
I think you want instance_eval here. Construct should be returning the newly created class, Esther than modifying the existing one, no?